karapace/backup/api.py

```diff
@@ -263,29 +262,53 @@ def _consumer(config: Config, topic: str) -> Iterator[KafkaConsumer]:
         yield consumer
 
 
-@contextlib.contextmanager
-def _enable_producer_callback_errors() -> Iterator[None]:
-    global_value = Future.error_on_callbacks
-    Future.error_on_callbacks = True
-    try:
-        yield None
-    finally:
-        Future.error_on_callbacks = global_value
+class BackupRestorationProducer:
```

Comment:
thought (non-blocking): I'm not a very big fan of this wrapper, it seems like we could have easily just bound parameters to a callback function inline inside `restore_backup` to achieve the same thing. That would have avoided having to echo the pretty big signature of the `.send()` method. No need to change though, this is somewhat opinionated.